lib/fs.js

```diff
@@ -1859,7 +1873,8 @@ WriteStream.prototype._write = function(data, encoding, cb) {
     });
 
   var self = this;
-  fs.write(this.fd, data, 0, data.length, this.pos, function(er, bytes) {
```

Comment:
This line is what this PR is all about: avoiding all the stuff in fs.write that we already know does not apply to this use-case.